pyatlan/client/asset.py

```diff
@@ -1421,3 +1422,152 @@ def _get_next_page(self):
+
+        :returns: a list of all the Assets that were created
+        """
+        return self._created
+
+    @property
+    def updated(self) -> list[Asset]:
+        """Get a list of all the Assets that were updated
+
+        :returns: a list of all the Assets that were updated
+        """
+        return self._updated
+
+    def add(self, single: Asset) -> Optional[AssetMutationResponse]:
+        """
+        Add an asset to the batch to be processed.
+
+        :param single: the asset to add to a batch
+        :returns: an AssetMutationResponse containing the results of the save or None if the batch is still queued.
+        """
+        self._batch.append(single)
+        return self._process()
+
+    def _process(self) -> Optional[AssetMutationResponse]:
+        """If the number of entities we have queued up is equal to the batch size, process them and reset our queue;
+        otherwise do nothing.
+
+        :returns: an AssetMutationResponse containing the results of the save or None if the batch is still queued.
+        """
+        if len(self._batch) == self._max_size:
+            return self.flush()
+        return None
+
+    def flush(self) -> Optional[AssetMutationResponse]:
+        """Flush any remaining assets in the batch.
+
+        :returns: n AssetMutationResponse containing the results of the saving any assets that were flushed
+        """
+        response: Optional[AssetMutationResponse] = None
+        if self._batch:
+            try:
+                if self._custom_metadata_handling == CustomMetadataHandling.IGNORE:
+                    response = self._client.save(
+                        self._batch, replace_atlan_tags=self._replace_atlan_tags
+                    )
+                elif self._custom_metadata_handling == CustomMetadataHandling.OVERWRITE:
+                    response = self._client.save_replacing_cm(
+                        self._batch, replace_atlan_tags=self._replace_atlan_tags
+                    )
+                elif self._custom_metadata_handling == CustomMetadataHandling.MERGE:
+                    response = self._client.save_merging_cm(
+                        self._batch, replace_atlan_tags=self._replace_atlan_tags
+                    )
+                else:
+                    raise ErrorCode.INVALID_PARAMETER_TYPE.exception_with_parameters(
+                        self._custom_metadata_handling,
+                        "CustomMetadataHandling.IGNORE, CustomMetadataHandling.OVERWRITE "
+                        "or CustomMetadataHandling.MERGE",
+                    )
+            except AtlanError as er:
+                if self._capture_failures:
+                    self._failures.append(
+                        FailedBatch(failed_assets=self._batch, failure_reason=er)
+                    )
+                else:
+                    raise er
+            self._batch = []
+        if response:
+            self._track_repsonse(response=response)
```

Comment:
Minor given its internal, but probably should fix the typo here: `_track_repsonse` -> `_track_response`

Reply:
done